### PR TITLE
Ambassador edge

### PR DESCRIPTION
**Pull request by slively**
So far I've really like using this over bare ambassador. The `Host` definition just works and it's awesome. We have a couple big downsides though.

See the different editions here: https://www.getambassador.io/editions/

We are no long using just open source ambassador but the community edition of the ambassador edge stack. It also has a requests per second limit on some functionality, but it doesn't seem that applies to inbound traffic. I have an email out to sales to talk about the limitations.

Basically:
1. I don't think we are using a fully open source product
2. It may have rate limits, but I don't think they effect us.

In all of our conversations we've really preferred open source solutions, and while this is mostly open source it doesn't quite fit the bill, but the extra ssl feature is pretty amazing and seems worth it. Assuming the rate limits don't impact us, it seems like a good path.

### CODE REVIEW

File: docs/AMBASSADOR.md [hunk 1]
```diff
@@ -1,10 +1,32 @@
 # Ambassador
 
-All of our cluster routing is handled by [Ambassador](https://www.getambassador.io/). If you like to debug the Amassador
-routing you must first find a running ambassador pod like:
+All inbound traffic to the MVP Studio cluster is handled by [Ambassador](https://www.getambassador.io/). It operates basically as an [Ingress Controller](https://kubernetes.io/docs/concepts/services-networking/ingress-controllers/), but we use the custom resources defined by Ambassador instead of the Ingress resource.
```
Comment (oliverdain) [inline docs/AMBASSADOR.md]: Here and below wrap lines to 120

File: docs/AMBASSADOR.md [hunk 2]
```diff
@@ -15,7 +37,7 @@ Obviously the actual pod names will likely be different when you run this. Once
 set up port forwarding:
 
 ```
-$ kubectl --namespace=ambassador port-forward ambassador-8db4cb58d-gsmjj 8877 8877
+$ kubectl --namespace=ambassador-edge port-forward ambassador-8db4cb58d-gsmjj 8877 8877
```
Comment (oliverdain) [inline docs/AMBASSADOR.md]: 2 things:

1. I just learned you can port-forward to a *deployment* and the deployment then picks the pod. That means you don't need to first list the pods, etc. just do a `kubectl port-forward deployment/ambassador ...`
2. You're forwarding the same port twice.

File: running/ambassador-edge/aes.yaml [hunk 3]
```diff
@@ -0,0 +1,332 @@
...
+  selector:
+    matchLabels:
+      service: ambassador-redis
+  template:
+    metadata:
+      labels:
+        service: ambassador-redis
+    spec:
+      containers:
+        - name: redis
```
Comment (oliverdain) [inline running/ambassador-edge/aes.yaml:126]: you're running redis but haven't given it any storage. Does it need to persist across restarts or is it just using it as a cache?

Comment (oliverdain) [reply running/ambassador-edge/aes.yaml:126]: Looks like it's fine - found this in the manual:

> The Ambassador Edge Stack relies on Redis to store short-lived authentication credentials and rate limiting information. If the Redis data store is lost, users will need to log back in and all existing rate-limits would be reset.

Maybe add that to the README so the next person doesn't have the same question.

**Review by oliverdain — APPROVED** [review]

File: docs/AMBASSADOR.md (hunk 2 above)
Comment (slively) [reply docs/AMBASSADOR.md]: oh that's nice

File: running/ambassador-edge/aes.yaml (hunk 3 above)
Comment (slively) [reply running/ambassador-edge/aes.yaml:126]: cool added a redis section to the readme

File: docs/AMBASSADOR.md (hunk 1 above)
Comment (slively) [reply docs/AMBASSADOR.md]: fixed everywhere, didn't have line wraps turned on in vscode
